services/visual_recognition/v3.js

```diff
@@ -0,0 +1,348 @@
+            {
+              // before pushing the function into the task array wrap the push in an IIFE function, passing in the 'prop' parameter
+              (function(prop, list_params, msg) {
+
+               asyncTasks.push(function (callback) {
+                  var buffer = msg.params[prop];
+                  temp.open({suffix: '.' + fileType(buffer).ext}, function (err, info) {
+                    if (err) {
+                      this.status({fill:'red', shape:'ring', text:'unable to open image stream'});          
+                      var message ='Node has been unable to open the image stream'; 
+                      node.error(message, msg);
+                      callback('open error on '+prop);
+                    }  
+                    stream_buffer(info.path, msg.params[prop], function () {
+                      list_params[prop]=fs.createReadStream(info.path);
+                      callback(null, prop);
+                    });
+                  }); // temp.open
+              }); // asyncTasks.push
+
+              })(prop, list_params, msg);
+
+            } else if (prop=='name') {
+              list_params[prop]=msg.params[prop];
+            }
+          } // for
+          
+
+          async.parallel(asyncTasks, function(error, results){
+            // when all temp local copies are ready, copy of all parameters and request to watson api
+            console.log(error, results);
+            if (error)
+            {
+              console.log("Parallel ended with error " + error);
+              return;
+            }
+            for (p in list_params)
+              params[p]=list_params[p];
+            performAction(params, feature, actionComplete2);
+          });
+      }
+        else if (feature=='retrieveClassifiersList') { 
+          performAction(params, feature, actionComplete2);
+      }
+        else if (feature =='retrieveClassifierDetails') {
+          params['classifier_id']=msg.params['classifier_id'];
+          performAction(params, feature, actionComplete2);
+      }
+        else if (feature=='deleteClassifier') {
+        params['classifier_id']=msg.params['classifier_id'];
+        performAction(params, feature, actionCompleteDeleteClassifier);
+
+      } else if (urlCheck(msg.payload)) {
+        params['url'] = msg.payload;
+        if ( msg.params != null && msg.params.classifier_ids != null)
+            params['classifier_ids']=msg.params['classifier_ids'];
+        if ( msg.params != null && msg.params.owners != null)
+            params['owners']=msg.params['owners'];
+        if ( msg.params != null && msg.params.threshold != null)
+            params['threshold']=msg.params['threshold'];
+        performAction(params, feature, actionComplete);
+      } else {
+        this.status({fill:'red', shape:'ring', text:'payload is invalid'});          
+        var message ='Payload must be either an image buffer or a string representing a url'; 
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Variable 'message' is already declared in line 148.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2686950655/issues/source?bid=3225115&fileBranchId=3338101#l328)